pkg/agent/controller/traceflow/packetin.go

```diff
@@ -237,7 +237,13 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*crdv1alpha1.Tracefl
 	if tableID == openflow.L2ForwardingOutTable.GetID() {
 		ob := new(crdv1alpha1.Observation)
 		tunnelDstIP := ""
-		isIPv6 := c.nodeConfig.NodeIPv6Addr != nil
+		var isIPv6 bool
+		// On dual stack, decide according to packet.
+		if c.nodeConfig.NodeIPv4Addr != nil && c.nodeConfig.NodeIPv6Addr != nil {
+			isIPv6 = etherData.Ethertype == protocol.IPv6_MSG
```

Comment:
Just confused that you already check the L3 protocol from the packet header, what this if-condition `c.nodeConfig.NodeIPv4Addr != nil && c.nodeConfig.NodeIPv6Addr != nil` is needed?

Reply:
Thanks for review.
You are correct, this condition check isn't required. Changed to decide based on packet only for any(ipv4/ipv6/dual) stack cluster.